pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -0,0 +1,1577 @@
+		/// - `Error::<T>::NoPermission`: If the caller is not the owner or
+		///   manager of the pool.
+		#[pallet::call_index(12)]
+		#[pallet::weight(T::WeightInfo::force_start_destroy(currency_count.to_owned()))]
+		pub fn force_start_destroy(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			currency_count: u32,
+		) -> DispatchResultWithPostInfo {
+			T::ForceOrigin::ensure_origin(origin)?;
+
+			let actual_currency_count = Self::do_start_destroy_pool(pool_id, currency_count, true, None)?;
+
+			Ok(Some(T::WeightInfo::force_start_destroy(actual_currency_count)).into())
+		}
+
+		/// Finishes the destruction process for a pool. The pool will be
+		/// removed from the storage and all leftover collateral and deposits
+		/// will be refunded to the owner.
+		///
+		/// # Parameters
+		/// - `origin`: The origin of the call.
+		/// - `pool_id`: The identifier of the pool to be destroyed.
+		/// - `currency_count`: The currency count in the pool, required for
+		///   weight calculation.
+		///
+		/// # Returns
+		/// - `DispatchResultWithPostInfo`: The result of the dispatch with the
+		///   actual used weights.
+		///
+		/// # Errors
+		/// - `Error::<T>::PoolUnknown`: If the pool does not exist.
+		/// - `Error::<T>::CurrencyCount`: If the number of currencies exceeds
+		///   `currency_count`.
+		/// - `Error::<T>::LivePool`: If the pool is not in a destroying state
+		///   or if there are any accounts left on any currency.
+		#[pallet::call_index(13)]
+		#[pallet::weight(T::WeightInfo::finish_destroy(currency_count.to_owned()))]
+		pub fn finish_destroy(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			currency_count: u32,
+		) -> DispatchResultWithPostInfo {
+			T::DefaultOrigin::ensure_origin(origin)?;
+
+			let pool_details = Pools::<T>::get(&pool_id).ok_or(Error::<T>::PoolUnknown)?;
+
+			let n_currencies = Self::get_currencies_number(&pool_details);
+
+			ensure!(n_currencies <= currency_count, Error::<T>::CurrencyCount);
+
+			ensure!(pool_details.state.is_destroying(), Error::<T>::LivePool);
+
+			for asset_id in pool_details.bonded_currencies {
+				if T::Fungibles::asset_exists(asset_id.clone()) {
+					// This would fail with an LiveAsset error if there are any accounts left on any
+					// currency
+					T::Fungibles::finish_destroy(asset_id)?;
+				}
+			}
+
+			let pool_account = pool_id.clone().into();
+
+			let total_collateral_issuance =
+				T::CollateralCurrencies::total_balance(pool_details.collateral_id.clone(), &pool_account);
+
+			if total_collateral_issuance > CollateralCurrenciesBalanceOf::<T>::zero() {
+				T::CollateralCurrencies::transfer(
+					pool_details.collateral_id,
+					&pool_account,
+					&pool_details.owner,
+					total_collateral_issuance,
+					Preservation::Expendable,
+				)?;
+			}
+
+			Pools::<T>::remove(&pool_id);
+
+			T::DepositCurrency::release(
+				&T::RuntimeHoldReason::from(HoldReason::Deposit),
+				&pool_details.owner,
+				Self::calculate_pool_deposit(n_currencies),
```

Comment:
Yeah Adel and I discussed this and kinda decided against supporting deposit amount configuration changes. 
Would be great to have a unified location for these deposit-paid memos if we want to rely on them for every storage map we keep, bundling them up with the storage items underwriting the proper function of a feature/pallet is incredibly ugly. Can your deposit pallet maybe support this?

Reply:
Yes, unfortunately it's not possible for the storage deposit pallet to implement the required trait, at least I did not find a way during the day or two I was playing around with it. There would need to be a `LifetimeHook` trait that is called before and after each key operation is executed, from pool creation, to minting/burning, to pool teardown. I could spend some time investigating whether I can make the deposit pallet work tho. But since we need hooks anyway to, for instance, return the amount of collateral that was used in a mint/burn, I would simply inject those hooks anyway, as they make the pallet much more flexible.

Reply:
talked to adel again and added the amount held in 85f93d65cda51d1d253a161469f2df4cc6696f04

Reply:
Unless we find a way to make the deposit storage pallet work (currently under research), why not follow the same approach we follow elsewhere? E.g., https://github.com/KILTprotocol/kilt-node/blob/fe579d61af3d442d4ba61f293feeea1402431486/pallets/attestation/src/lib.rs#L567.

Reply:
This would still require us to store the deposit amount along with the pool details, right? For a standard use case like the one here, which makes no distinction between an 'owner' of a storage entry and a depositor, this seems overkill... The hook approach appears more flexible but is also way more complex and less 'standard' than just relying on the fungibles trait, while being functionally identical as far as I can tell. Personally, I'd rather not switch if not required

Reply:
I suggested that since it would mean we use the same approach everywhere when it comes to handling deposits. Hopefully soon, all this can be switched with the deposit storage pallet (WIP), but sure, whatever you guys think it's ok.